src/loader/loader.js

```diff
@@ -290,27 +309,37 @@ export default class Loader {
 		}
 
 		return setTimeout(() => {
-			const registeredModules = config.getModules();
 			const resolvedModuleNames = resolution.resolvedModules;
 
-			const missingDependencies = resolvedModuleNames.filter(
-				dep =>
-					registeredModules[dep] === undefined ||
-					registeredModules[dep].implementation === undefined
-			);
+			const missingDependencies = resolvedModuleNames.filter(dep => {
+				const module = config.getModule(dep);
+				return !module || !module.implemented;
+			});
 
 			const fetchedMissingDependencies = missingDependencies.filter(
-				dep =>
-					registeredModules[dep] !== undefined &&
-					registeredModules[dep].fetched
+				dep => {
+					const module = config.getModule(dep);
+					return module && module.fetched;
+				}
 			);
 
 			const unfetchedMissingDependencies = missingDependencies.filter(
-				dep =>
-					registeredModules[dep] !== undefined &&
-					!registeredModules[dep].fetched
+				dep => {
+					const module = config.getModule(dep);
+					return module && !module.fetched;
```

Comment:
Should this be `!module || !module.fetched`? Asking because I would expect `fetchedMissingDependencies` + `unfetchedMissingDependencies` to equal `missingDependencies`, but it appears that any `module` that is falsey won't appear in either of "fetched" or "unfetched".

Reply:
That's a good question. The code is OK because in the end it does what it is supposed to do, but I think it is because we are making assumptions about things happenning in other places while writing this filters. I will give it a thought and enhance the code.

Reply:
The thing is that all modules are defined in this point of the code because they are defined before writing the <script> tags to the page so, once you arrive here, they are defined for sure.

Reply:
I've pushed one commit.